roofit/multiprocess/src/JobManager.cxx

```diff
@@ -142,9 +153,9 @@ Messenger &JobManager::messenger() const
    return *messenger_ptr_;
 }
 
-Queue &JobManager::queue() const
+Queue *JobManager::queue() const
```

Comment:
Why is this returning a pointer now? Can the `queue_ptr_` be null?

Reply:
Because the interface of the Queue subclasses is not uniform. I have a dynamic_cast somewhere down the line. It doesn't make sense to define an abstract interface in the base class, because the FIFO queue simply lacks the functionality for priority/ordering.

I suspect there are better possible designs for this, I'm open to suggestions, but maybe we shouldn't overengineer at this point, because there will be another variant upcoming that may again change considerations.

Reply:
Alright, no need to over-engineer of course, I just curious :+1: